python/pysmurf/core/utilities/_SmurfPublisher.py

```diff
@@ -196,4 +205,27 @@ def register_file(self, path, type, format='', timestamp=None, plot=False):
         return self.publish(file_data, 'data_file')
 
     def _varListen(self, path, varVal):
-        self.publish(data=f'{path}={varVal.value}',msgtype='general')
+        """
+        Callback function used to publish metadata.
+
+        Args
+        ----
+        path : str
+            Rogue register path.
+        varVal : pyrogue.VariableValue
+            Variable object.
+        """
+        # Extract the variable value
+        value = varVal.value
+
+        # Get the variable type, as a string
+        type_str = type(value).__name__
+
+        # In the variable is a numpy array, convert it to a list in
+        # order to be JSON serializable (but keeping the original type)
+        if isinstance(value, np.ndarray):
+            value = value.tolist()
```

Comment:
When I experienced issue #594 it seemed like strings were actually being returned as a np.ndarray of chars. Does this sound familiar to you? If possible I think we should convert these to actual strings, but I don't think it's a huge deal if these have to be converted in the pysmurf-monitor, as long as the type information is correct.

Reply:
That conversion of some type of variables (like `np.ndarray`, or integers greater than 32-bit, for example) happen only throng the EPICS interface (i.e. when the variable is converted to an EPICS PV), due to limitations on the type of variable EPICS supports. 
In this case, on the other hand, this are local variables (i.e. they are being processed in the same Rogue root instance), so that conversion does not happen.